dbms/src/Storages/StorageDistributed.cpp

```diff
@@ -133,6 +137,29 @@ void initializeFileNamesIncrement(const std::string & path, SimpleIncrement & in
         increment.set(getMaximumFileNumber(path));
 }
 
+/// the same as DistributedBlockOutputStream::createSelector, should it be static?
+IColumn::Selector createSelector(const ClusterPtr cluster, const ColumnWithTypeAndName & result)
+{
+    const auto & slot_to_shard = cluster->getSlotToShard();
+
+#define CREATE_FOR_TYPE(TYPE)                                   \
+    if (typeid_cast<const DataType##TYPE *>(result.type.get())) \
```

Comment:
It's not the same. See https://github.com/yandex/ClickHouse/pull/3683
Better to remove copy-paste.